Neos.Flow/Classes/Security/Context.php

```diff
@@ -639,7 +639,7 @@ protected function collectRolesAndParentRolesFromAccount(Account $account): arra
             return $roles;
         };
 
-        return array_reduce($account->getRoles(), $reducer, []);
+        return array_reduce(iterator_to_array($account->getRoles()->getIterator()), $reducer, []);
```

Comment:
That screams for a convenience method in the `Roles` value object so that the "iterater" low level logic could be moved there. Sth like:
```php
return $account->getRoles()->merge($parentRoles);
```